### PR TITLE
feat(management): add K8sAuth security scheme for X-Kubernetes-Authorization

### PR DESCRIPTION
**Pull request by Matovidlo**
**Changes:**
- Add `K8sAuth` apiKey security scheme (`X-Kubernetes-Authorization` header) to the Manage API OpenAPI spec
- Allow `K8sAuth` alongside `ApiKeyAuth` on `GET /manage/tokens/verify`
- Regenerate `api_token_verification.go` with openapi-generator 7.19.0 (only this operation changed; manual fixes in `client.go`/models preserved)
- Add `X-Kubernetes-Authorization` to the redacted-headers list in debug logging
- Export auth scheme and header name constants (`AuthSchemeAPIKey`, `AuthSchemeK8s`, `HeaderManageAPIToken`, …)
- Add per-request `Auth` strategy (`ManageAPITokenAuth`, `KeboolaServiceAccountAuth`, `NewAutoAuth`) with SA token auto-rotation, applied via `NewAPIClientWithAuth` — matches the keboola-operator and PHP client approach
- `NewAPIClientWithAuth` rejects a nil `Auth` with an error (fail-fast at construction) and the auth transport enforces `Auth` precedence over `ContextAPIKeys` headers
- Add unit tests for both auth schemes, token rotation, nil-auth, and the `Auth`-over-`ContextAPIKeys` precedence contract

-----------

Linear: https://linear.app/keboola/issue/PSGO-256/add-support-for-manage-sdk-in-go-monorepo

Context: `git-service` in go-monorepo currently hand-rolls the HTTP request for Kubernetes token verification because the generated client only knew `ApiKeyAuth` (keboola/go-monorepo#524). With this change the SDK covers both paths via the `Auth` strategy:

```go
// Auto-configuration: non-empty token -> ManageAPITokenAuth,
// empty token -> KeboolaServiceAccountAuth (projected SA token,
// re-read from disk on every request so rotation is picked up):
auth, err := management.NewAutoAuth(os.Getenv("KBC_MANAGE_TOKEN"))
client, err := management.NewAPIClientWithAuth(management.NewConfiguration(), auth)
resp, httpResp, err := client.TokenVerificationAPI.TokenVerification(ctx).Execute()
```

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Review by pepamartinec — CHANGES_REQUESTED** [review]
* ten K8S token se prubezne rotuje, takze by nemel byt staticky v konfiguraci klienta
* kde/jak ten token vemu? (pominme, ze konkretne ja bych to vedel)
* skvela by byla auto-konfigurace (pokud nezadam standardni manage-token, pouzije se SA token ze standardni cesty)

Mrkni se, jak to je klientovi v operatoru (https://github.com/keboola/keboola-operator/blob/4db25eabed066b0071fcf6bf8aa8e7608665fb47/internal/client/manageapi/client.go#L70). V PHP klientech to resime stejne (https://github.com/keboola/platform-libraries/blob/503657051bd665f9abcf9d5d82eeaeeaa563c15b/libs/git-service-api-client/src/Auth/AuthInterface.php#L7)

**Comment by Matovidlo** [comment]
@pepamartinec Done in 2ffeda5 — adopted the operator pattern (`internal/client/manageapi/auth.go`):

- **`Auth` interface** (`AuthHeaders() (map[string]string, error)`) resolved on **every request**, so the rotated SA token is picked up automatically — `KeboolaServiceAccountAuth` re-reads the file per request and sends `Bearer <token>`, defaulting to `/var/run/secrets/connection.keboola.com/serviceaccount/token`.
- **Auto-configuration** via `NewAutoAuth(token)` — non-empty token → `ManageAPITokenAuth`, empty → SA token from the standard path.
- Wired into the generated client through an `http.RoundTripper` (`NewAPIClientWithAuth`), so no generated files are touched and nothing static lives in the client config.

```go
auth, err := management.NewAutoAuth(os.Getenv("KBC_MANAGE_TOKEN"))
client := management.NewAPIClientWithAuth(management.NewConfiguration(), auth)
```

Covered by tests incl. a rotation test (token file rewritten between two requests). The static `ContextAPIKeys` path stays as a low-level fallback; the `Auth` strategy is the documented/recommended way.

File: pkg/keboola/management/README.md [hunk 1]
```diff
@@ -454,6 +462,56 @@ auth := context.WithValue(
 r, err := client.Service.Operation(auth, args)
 ```
 
+### K8sAuth
```
Comment (pepamartinec) [inline pkg/keboola/management/README.md]: **Drop this entire `### K8sAuth` section** (down to the `### Recommended` heading).

Nobody should use this low-level path; all K8s auth should go through the `Auth` strategy (`NewAPIClientWithAuth`).

File: pkg/keboola/management/auth_transport.go [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+// Note: the generated callAPI dumps the request for debug logging before the
+// transport runs, so headers set here never appear in the debug dump and the
+// token is never logged.
+type authTransport struct {
+	base http.RoundTripper // nil falls back to http.DefaultTransport
+	auth Auth
+}
+
+func (t *authTransport) RoundTrip(req *http.Request) (*http.Response, error) {
+	headers, err := t.auth.AuthHeaders()
```
Comment (pepamartinec) [inline pkg/keboola/management/auth_transport.go:22]: (nit) `auth` is dereferenced unconditionally, so `NewAPIClientWithAuth(cfg, nil)` panics on the first request rather than failing fast. Consider a nil-check in `NewAPIClientWithAuth` (return an error, or skip wrapping the transport), or at least documenting that `auth` must be non-nil.

File: pkg/keboola/management/auth_transport.go [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+		base = http.DefaultTransport
+	}
+
+	return base.RoundTrip(req)
+}
+
+// NewAPIClientWithAuth creates an APIClient that resolves authentication
+// headers via the given Auth on every request, so rotating credentials
+// (e.g. a projected Kubernetes ServiceAccount token) are picked up
+// automatically. When an Auth is installed it takes precedence over headers
```
Comment (pepamartinec) [inline pkg/keboola/management/auth_transport.go:72]: (optional) This documented precedence — `Auth` over `ContextAPIKeys` — holds, since the transport runs last and uses `Header.Set`. But no test sets *both* mechanisms at once; a small test would lock the contract in.

**Review by pepamartinec — COMMENTED** [review]
Jeste par dropbnosti

File: pkg/keboola/management/README.md (hunk 1 above)
Comment (Matovidlo) [reply pkg/keboola/management/README.md]: **Fixed** ✅ in commit e232c50348fb62fc6b12e767300fcb1885a87a6d

Dropped the entire `### K8sAuth` section down to `### Recommended` and reworded the intro of the dynamic-auth section so it no longer references the removed low-level path. All K8s auth now documented exclusively through the `Auth` strategy (`NewAPIClientWithAuth`).

File: pkg/keboola/management/auth_transport.go (hunk 2 above)
Comment (Matovidlo) [reply pkg/keboola/management/auth_transport.go:22]: **Fixed** ✅ in commit 860c21a06e94616822879d91ab9fd1173db95504

`NewAPIClientWithAuth` now returns `(*APIClient, error)` and rejects a nil `Auth` with `"auth must not be nil"`, so misconfiguration fails at construction instead of panicking on the first request. Added `Test_management_Auth_NilAuth` and updated all call sites + README example.

File: pkg/keboola/management/auth_transport.go (hunk 3 above)
Comment (Matovidlo) [reply pkg/keboola/management/auth_transport.go:72]: **Fixed** ✅ in commit 860c21a06e94616822879d91ab9fd1173db95504

Added `Test_management_Auth_TakesPrecedenceOverContextAPIKeys` — and it caught a real bug: the documented precedence did NOT hold. The generated `prepareRequest` assigns header map keys without canonicalization (`headers[h] = []string{v}`), so the transport's `Header.Set` created a second, canonical key and BOTH values went out on the wire. The transport now drops case-insensitive duplicates before setting the Auth header; the test locks the contract in.

File: pkg/keboola/management/auth_transport.go [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+		// The generated prepareRequest assigns header map keys directly,
+		// bypassing canonicalization — drop any case-insensitive duplicates
+		// (e.g. from ContextAPIKeys) so the Auth value is the only one sent.
+		for existing := range req.Header {
+			if strings.EqualFold(existing, k) {
+				delete(req.Header, existing)
+			}
+		}
+		req.Header.Set(k, v)
+	}
```
[the remark this answers is not in the document]
Comment (Matovidlo) [reply pkg/keboola/management/auth_transport.go:42]: **Fixed** ✅ in commit ce7b7d7028fa4b404d9733f39b73993796b06ba8

Good catch — the transport now strips ALL known auth headers (`X-KBC-ManageApiToken` and `X-Kubernetes-Authorization`, case-insensitively via `isKnownAuthHeader`) before applying the Auth headers, so the installed Auth is the only auth mechanism on the wire regardless of which scheme `ContextAPIKeys` targeted. Extended `Test_management_Auth_TakesPrecedenceOverContextAPIKeys` to set BOTH schemes via `ContextAPIKeys` and assert the cross-scheme header is stripped.

File: pkg/keboola/management/test/token_verification_headers_test.go [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+		management.AuthSchemeAPIKey: {Key: "manage-token"},
+	})
+
+	resp, httpRes, err := newTestClient(server.URL).TokenVerificationAPI.TokenVerification(ctx).Execute()
+	require.NoError(t, err)
+	require.NotNil(t, resp)
+	assert.Equal(t, http.StatusOK, httpRes.StatusCode)
+	assert.Equal(t, "manage-token", gotHeader.Get(management.HeaderManageAPIToken))
+	assert.Empty(t, gotHeader.Get(management.HeaderKubernetesAuthorization))
+	assert.Equal(t, []interface{}{"git-repo:manage"}, resp.GetScopes())
```
[the remark this answers is not in the document]
Comment (Matovidlo) [reply pkg/keboola/management/test/token_verification_headers_test.go]: **Fixed** ✅ in commit 1f6c24631c66e3156df7193f0a00fb3f18e79522

Replaced the inline server with the shared `newAuthTestServer` helper, which now captures headers under a `sync.Mutex` and returns a synchronized accessor — no shared-variable write from the handler goroutine remains. Verified with `go test -race -count=1 ./pkg/keboola/management/test/`.

File: pkg/keboola/management/test/token_verification_headers_test.go [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+		management.AuthSchemeK8s: {Key: "k8s-token"},
+	})
+
+	resp, httpRes, err := newTestClient(server.URL).TokenVerificationAPI.TokenVerification(ctx).Execute()
+	require.NoError(t, err)
+	require.NotNil(t, resp)
+	assert.Equal(t, http.StatusOK, httpRes.StatusCode)
+	assert.Equal(t, "k8s-token", gotHeader.Get(management.HeaderKubernetesAuthorization))
+	assert.Empty(t, gotHeader.Get(management.HeaderManageAPIToken))
+	assert.Equal(t, []interface{}{"git-repo:manage"}, resp.GetScopes())
```
[the remark this answers is not in the document]
Comment (Matovidlo) [reply pkg/keboola/management/test/token_verification_headers_test.go]: **Fixed** ✅ in commit 1f6c24631c66e3156df7193f0a00fb3f18e79522

Same fix as the test above — this inline server was also replaced with the mutex-synchronized `newAuthTestServer` helper (both files are in the same test package). Verified under `go test -race`.

File: pkg/keboola/management/test/auth_test.go [hunk 7]
```diff
@@ -0,0 +1,191 @@
...
+)
+
+// newAuthTestServer returns a test server that records the headers of the
+// last request and responds with a valid token verification payload.
+func newAuthTestServer(headers *http.Header) *httptest.Server {
+	return httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		*headers = r.Header.Clone()
+		w.Header().Set("Content-Type", "application/json")
+		_, _ = w.Write([]byte(`{"scopes": ["git-repo:manage"]}`))
+	}))
```
[the remark this answers is not in the document]
Comment (Matovidlo) [reply pkg/keboola/management/test/auth_test.go]: **Fixed** ✅ in commit 1f6c24631c66e3156df7193f0a00fb3f18e79522

`newAuthTestServer` no longer writes through a shared `*http.Header`: it guards the capture with a `sync.Mutex` and returns a `func() http.Header` accessor that takes the same lock, so reads in the test goroutine are race-free by construction. All call sites updated; `go test -race -count=1` passes.

File: pkg/keboola/management/README.md [hunk 8]
```diff
@@ -454,6 +462,38 @@ auth := context.WithValue(
 r, err := client.Service.Operation(auth, args)
 ```
 
+### Recommended: dynamic auth (rotating tokens)
```
[the remark this answers is not in the document]
Comment (Matovidlo) [reply pkg/keboola/management/README.md:465]: **Alternative approach** 💡

Re-adding the `### K8sAuth` section would contradict the explicit review request to drop it (https://github.com/keboola/keboola-sdk-go/pull/89#discussion_r3361239059) — the static `ContextAPIKeys` path can't follow SA token rotation and shouldn't be advertised. Instead, the broken anchor in `docs/TokenVerificationAPI.md` now points to `#recommended-dynamic-auth-rotating-tokens`, which documents the supported way to configure K8s auth.

**Review by pepamartinec — APPROVED** [review]
mrkni jeste, co pise Copilot, ja uz vic nemam 👍
